pkg/app/piped/controller/planner.go

```diff
@@ -97,6 +97,7 @@ func newPlanner(
 		pipedConfig:              pipedConfig,
 		plannerRegistry:          registry.DefaultRegistry(),
 		appManifestsCache:        appManifestsCache,
+		cancelledCh:              make(chan *model.ReportableCommand, 1),
```

Comment:
I'm very curious about why using buffered-channel. Is there something wrong to use an unbuffered channel with zero capacity?

Reply:
Fine, I'm getting get it.

Reply:
This is for preventing the `Cancel()` from being blocked until Run handles the channel, right?
https://github.com/pipe-cd/pipe/blob/f1cee728d7764846f58ef29d4641f96e6dfb1900/pkg/app/piped/controller/planner.go#L201-L206

Reply:
Right or not. 

Currently, when the planner finished its work (`Run`), the `cancelledCh` will not be used after that.
So in case the piped receives a canceling command immediately after the planner's completion, the controller will be blocked forever.

Reply:
That makes more sense! Now I can see why using it at this time, thanks :D